src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -137,7 +137,7 @@ public function indexAction(Request $request, ProductFilters $filters): Response
             'categoryFilterForm' => $categoriesForm->createView(),
             'productGrid' => $this->presentGrid($productGrid),
             'enableSidebar' => true,
-            'layoutHeaderToolbarBtn' => $this->getProductToolbarButtons(),
+            'layoutHeaderToolbarBtn' => $this->getProductToolbarButtons($request->get('_legacy_controller')),
```

Comment:
Like so ?

```suggestion
            'layoutHeaderToolbarBtn' => $this->getProductToolbarButtons(Permission::CREATE, $request->get('_legacy_controller')),
```

Reply:
Better check the permission in the method itself, for now, there's only one button but if others are added they may be linked to another permission, the controller associated with this page however should not change.